reflex/model.py

```diff
@@ -51,7 +51,99 @@ def get_engine(url: str | None = None):
     return sqlmodel.create_engine(url, echo=echo_db_query, connect_args=connect_args)
 
 
-class Model(Base, sqlmodel.SQLModel):
+SQLModelOrSqlAlchemy = Union[
+    Type[sqlmodel.SQLModel], Type[sqlalchemy.orm.DeclarativeBase]
+]
+
+
+class ModelRegistry:
+    """Registry for all models."""
+
+    models: ClassVar[set[SQLModelOrSqlAlchemy]] = set()
+
+    @classmethod
+    def register(cls, model: SQLModelOrSqlAlchemy):
+        """Register a model. Can be used directly or as a decorator.
+
+        Args:
+            model: The model to register.
+
+        Returns:
+            The model passed in as an argument (Allows decorator usage)
+        """
+        cls.models.add(model)
+        return model
+
+    @classmethod
+    def get_models(cls, include_empty: bool = False) -> set[SQLModelOrSqlAlchemy]:
+        """Get registered models.
+
+        Args:
+            include_empty: If True, include models with empty metadata.
+
+        Returns:
+            The registered models.
+        """
+        if include_empty:
+            return cls.models
+        return {
+            model for model in cls.models if not cls._model_metadata_is_empty(model)
+        }
+
+    @staticmethod
+    def _model_metadata_is_empty(model: SQLModelOrSqlAlchemy) -> bool:
+        """Check if the model metadata is empty.
+
+        Args:
+            model: The model to check.
+
+        Returns:
+            True if the model metadata is empty, False otherwise.
+        """
+        return len(model.metadata.tables) == 0
+
+    @classmethod
+    @property
+    def metadata(cls) -> sqlalchemy.MetaData:
```

Comment:
Looks like joining these isn't supported in 3.8 (and won't be supported in 3.13 again later) which is why the unit tests are failing: https://stackoverflow.com/questions/128573/using-property-on-classmethods/64738850#64738850

Reply:
Thanks for the hint, I refactored it to not use chained classmethods and propertys anymore. I also refactored the `Model.select` method, which chained those decorators as well. However, that was not the reason for the tests to fail.  

I cleaned everything up now, fixed the tests and rebased on main.